File: frontend/src/app/main-module/models/subscription.ts
import {SubscriptionCategory} from "./subscription-category";

export class Subscription {
    id: string;
    name: string;
    imageUrl: string;
    description: string;
    price: number;
    ownerId: string;
    category: SubscriptionCategory;

    static cloneSubscription(subscription: Subscription): Subscription {
        let clonedSubscription: Subscription = new Subscription();
        clonedSubscription.id = subscription.id;
        clonedSubscription.name = subscription.name;
        clonedSubscription.imageUrl = subscription.imageUrl;
        clonedSubscription.description = subscription.description;
        clonedSubscription.price = subscription.price;
        clonedSubscription.ownerId = subscription.ownerId;
        if (subscription.category) {
            clonedSubscription.category = SubscriptionCategory.cloneCategory(subscription.category);
        }
        return clonedSubscription;
    }
}

File: apigateway/src/main/java/ru/otus/apigateway/model/backend/Subscription.java
package ru.otus.apigateway.model.backend;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.otus.apigateway.model.view.CategoryViewModel;
import ru.otus.apigateway.model.view.OwnerViewModel;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Subscription {

    private Long id;
    private Integer price;
    private String name;
    private String imageUrl;
    private String description;
    private Long ownerId;
    private Long categoryId;

    private OwnerViewModel owner;
    private CategoryViewModel category;
}


